src/size.rs

```diff
@@ -7,117 +7,158 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
-use length::Length;
+use length::{ Length, Untyped };
+use scale_factor::{ ScaleFactor };
```

Comment:
nit: no spaces after `{` and before `}`, and also no need to brace `{ScaleFactor}`